crates/types/src/net/mod.rs

```diff
@@ -78,10 +78,19 @@ fn parse_http(s: &str) -> Result<AdvertisedAddress, Error> {
         .parse::<Uri>()
         .with_context(|| format!("Invalid URI format: '{s}'"))?;
 
+    let mut parts = uri.into_parts();
+    // default to http if scheme is missing
+    if parts.scheme.is_none() {
+        parts.scheme = Some(http::uri::Scheme::HTTP);
+    }
```

Comment:
Sweet! Could we also default port to `5122` the same way, if omitted? Maybe unconventional if default scheme is http though.

Reply:
I wanted to do that but it's slightly more involved to do this.